app/app/Http/Controllers/LocationSearchController.php

```diff
@@ -400,4 +400,16 @@ public function search(Request $request)
 				'search_radius'      => $search_radius
 			]);
 	}
+	/**
+	 * set search radius
+	 */
+	public function setSearchRadius(Request $request) {
+		$distance = Input::get('distance');
+		if(is_numeric($distance)) {
+			$f_distance = floatval( $distance );
+			if(($f_distance < 200) && ($f_distance > 0))
```

Comment:
This 0 to 200km may seem like a reasonable range but it introduces some behaviour that isn't clear only from the requirements.  Greater than 0 seems pretty obvious, though.

Replace with just: "if ( $f_distance > 0 )".

Also, don't quietly ignore a request.  Instead, respond with an HTTP error code like you'd otherwise do with an exception. For this, add an else like this:
```
if ( $f_distance > 0 )
    BaseUser::setSearchRadius($f_distance);
else
    return Response::json([
    'message' => 'radius must be greater than 0'
], 422);
```